### ISSUE TITLE
service.dead fails when service is not installed.

### ISSUE THREAD

**Issue by sketch242** [saltstack/salt#38696]
service.dead fails when the service is not installed.  I have the following state:

```
postfix:
  service.dead:
    - enable: False
  pkg.removed:
    - pkgs:
        - postfix
```

The first time I run it, it is clean, but the second time around, it fails because the postfix service has been uninstalled.  If I replace service.dead with service.disabled, it does not return an error when the service is not installed, so this seems like a bug in service.dead.

testhost:
----------
          ID: postfix
    Function: service.dead
      Result: False
     Comment: The named service postfix is not available
     Started: 18:47:24.563870
    Duration: 19.276 ms
     Changes:   
  Name: postfix - Function: pkg.removed - Result: Clean Started: - 18:48:56.432033 Duration: 2369.843 ms

This is on CentOS 7, master and minion are both 2016.3.4.

**Comment by sketch242** [comment]
Just in case it's useful for anyone, here is my current workaround:

postfix:
{% if salt['service.available']('postfix') %}
  service.dead:
    - enable: False
{% endif %}
  pkg.removed:
    - pkgs:
        - postfix

**Comment by lorengordon** [comment]
I think this might be a [duplicate](https://github.com/saltstack/salt/issues/37511).

It looks like the fix was [merged into the 2016.3 branch](https://github.com/saltstack/salt/pull/37562). Are you able to give that a try to see if it fixes the problem for you?

**Comment by sketch242** [comment]
Yep, looks like it does.  I searched for similar issues for submitting mine, but I forgot to remove is:open, so I didn't see that one.  Thanks.
